core/templates/rid_owner.h

```diff
@@ -239,7 +244,7 @@ class RID_Alloc : public RID_AllocBase {
 
 		uint32_t validator = uint32_t(id >> 32);
 
-		bool owned = (validator_chunks[idx_chunk][idx_element] & 0x7FFFFFFF) == validator;
+		bool owned = (validator < 0x7FFFFFFF) && (validator_chunks[idx_chunk][idx_element] & 0x7FFFFFFF) == validator;
 
```

Comment:
By definition if the validator is `0x8000000` or more it cannot pass the second check, so the first check is effectively just for the value `0x7FFFFFFF` I think. It's debatable whether this is worth doing, as it's just adding a comparison to test for a super rare corrupted value I think.

And actually come to think of it, the `validator_chunks` should never get set to this (`0x7FFFFFF`) in the first place (due to the `CRASH_NOW`) so this check is redundant I suspect?

Reply:
Added to catch user input, but can remove

Reply:
But regardless of user input, if the `validator_chunks` are always correct, bad input will always be caught (as it won't equal the stored validator I think?).

Reply:
Will take a deeper look soon! Thank you